plugin/src/main/java/org/opensearch/ml/action/unload/TransportUnloadModelAction.java

```diff
@@ -128,7 +139,11 @@ protected UnloadModelNodesResponse newResponse(
                     BulkRequest bulkRequest = new BulkRequest();
                     for (String modelId : removedNodeMap.keySet()) {
                         UpdateRequest updateRequest = new UpdateRequest();
-                        updateRequest.index(ML_MODEL_INDEX).id(modelId).doc(ImmutableMap.of(MODEL_STATE_FIELD, MLModelState.UNLOADED));
+                        int removedNodeCount = removedNodeMap.get(modelId).size();
+                        MLModelState mlModelState = modelWorkNodeCounts.get(modelId) > removedNodeCount
```

Comment:
No need to check if removedNodeCount > 0 here?  
modelWorkNodeCounts.get(modelId) > removedNodeCount && removedNodeCount > 0.

Reply:
removedNodeCount will always > 0, check line 121